Watching for file changes with StatReloader
Watching for file changes with StatReloader
/home/rumi/Desktop/projects/my_jobs_env/django-job-portal/jobsapp/models.py changed, reloading.
/home/rumi/Desktop/projects/my_jobs_env/django-job-portal/jobsapp/models.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Unauthorized: /api/login/
Unauthorized: /api/login/
"POST /api/login/ HTTP/1.1" 401 151
"POST /api/login/ HTTP/1.1" 401 151
"POST /api/login/ HTTP/1.1" 200 873
"POST /api/login/ HTTP/1.1" 200 873
Bad Request: /api/apply-job/1
Bad Request: /api/apply-job/1
"POST /api/apply-job/1 HTTP/1.1" 400 150
"POST /api/apply-job/1 HTTP/1.1" 400 150
Bad Request: /api/apply-job/1
Bad Request: /api/apply-job/1
"POST /api/apply-job/1 HTTP/1.1" 400 150
"POST /api/apply-job/1 HTTP/1.1" 400 150
Internal Server Error: /api/apply-job/1
Traceback (most recent call last):
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/utils.py", line 86, in _execute
    return self.cursor.execute(sql, params)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/sqlite3/base.py", line 396, in execute
    return Database.Cursor.execute(self, query, params)
sqlite3.IntegrityError: NOT NULL constraint failed: jobsapp_applicant.user_id

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/views.py", line 505, in dispatch
    response = self.handle_exception(exc)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/views.py", line 465, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/views.py", line 476, in raise_uncaught_exception
    raise exc
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/views.py", line 502, in dispatch
    response = handler(request, *args, **kwargs)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/generics.py", line 190, in post
    return self.create(request, *args, **kwargs)
  File "/home/rumi/Desktop/projects/my_jobs_env/django-job-portal/jobsapp/api/views.py", line 38, in create
    self.perform_create(serializer)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/mixins.py", line 24, in perform_create
    serializer.save()
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/serializers.py", line 212, in save
    self.instance = self.create(validated_data)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/serializers.py", line 948, in create
    instance = ModelClass._default_manager.create(**validated_data)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/query.py", line 433, in create
    obj.save(force_insert=True, using=self.db)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/base.py", line 746, in save
    force_update=force_update, update_fields=update_fields)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/base.py", line 784, in save_base
    force_update, using, update_fields,
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/base.py", line 887, in _save_table
    results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/base.py", line 926, in _do_insert
    using=using, raw=raw,
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/query.py", line 1204, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/sql/compiler.py", line 1392, in execute_sql
    cursor.execute(sql, params)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/utils.py", line 100, in execute
    return super().execute(sql, params)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/utils.py", line 68, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/utils.py", line 77, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/utils.py", line 86, in _execute
    return self.cursor.execute(sql, params)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/utils.py", line 86, in _execute
    return self.cursor.execute(sql, params)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/sqlite3/base.py", line 396, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.IntegrityError: NOT NULL constraint failed: jobsapp_applicant.user_id
Internal Server Error: /api/apply-job/1
Traceback (most recent call last):
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/utils.py", line 86, in _execute
    return self.cursor.execute(sql, params)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/sqlite3/base.py", line 396, in execute
    return Database.Cursor.execute(self, query, params)
sqlite3.IntegrityError: NOT NULL constraint failed: jobsapp_applicant.user_id

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/views.py", line 505, in dispatch
    response = self.handle_exception(exc)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/views.py", line 465, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/views.py", line 476, in raise_uncaught_exception
    raise exc
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/views.py", line 502, in dispatch
    response = handler(request, *args, **kwargs)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/generics.py", line 190, in post
    return self.create(request, *args, **kwargs)
  File "/home/rumi/Desktop/projects/my_jobs_env/django-job-portal/jobsapp/api/views.py", line 38, in create
    self.perform_create(serializer)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/mixins.py", line 24, in perform_create
    serializer.save()
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/serializers.py", line 212, in save
    self.instance = self.create(validated_data)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/rest_framework/serializers.py", line 948, in create
    instance = ModelClass._default_manager.create(**validated_data)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/query.py", line 433, in create
    obj.save(force_insert=True, using=self.db)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/base.py", line 746, in save
    force_update=force_update, update_fields=update_fields)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/base.py", line 784, in save_base
    force_update, using, update_fields,
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/base.py", line 887, in _save_table
    results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/base.py", line 926, in _do_insert
    using=using, raw=raw,
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/query.py", line 1204, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/models/sql/compiler.py", line 1392, in execute_sql
    cursor.execute(sql, params)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/utils.py", line 100, in execute
    return super().execute(sql, params)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/utils.py", line 68, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/utils.py", line 77, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/utils.py", line 86, in _execute
    return self.cursor.execute(sql, params)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/utils.py", line 86, in _execute
    return self.cursor.execute(sql, params)
  File "/home/rumi/Desktop/projects/my_jobs_env/lib/python3.6/site-packages/django/db/backends/sqlite3/base.py", line 396, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.IntegrityError: NOT NULL constraint failed: jobsapp_applicant.user_id
"POST /api/apply-job/1 HTTP/1.1" 500 24108
"POST /api/apply-job/1 HTTP/1.1" 500 24108
/home/rumi/Desktop/projects/my_jobs_env/django-job-portal/jobsapp/api/serializers.py changed, reloading.
/home/rumi/Desktop/projects/my_jobs_env/django-job-portal/jobsapp/api/serializers.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Watching for file changes with StatReloader
"POST /api/apply-job/1 HTTP/1.1" 201 59
"POST /api/apply-job/1 HTTP/1.1" 201 59
Bad Request: /api/apply-job/1
Bad Request: /api/apply-job/1
"POST /api/apply-job/1 HTTP/1.1" 400 150
"POST /api/apply-job/1 HTTP/1.1" 400 150
Watching for file changes with StatReloader
Watching for file changes with StatReloader
/home/rumi/Desktop/projects/my_jobs_env/django-job-portal/jobsapp/api/views.py changed, reloading.
/home/rumi/Desktop/projects/my_jobs_env/django-job-portal/jobsapp/api/views.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Watching for file changes with StatReloader
/home/rumi/Desktop/projects/my_jobs_env/django-job-portal/jobsapp/api/views/employee.py changed, reloading.
/home/rumi/Desktop/projects/my_jobs_env/django-job-portal/jobsapp/api/views/employee.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
/home/rumi/Desktop/projects/my_jobs_env/django-job-portal/jobsapp/api/views/employee.py changed, reloading.
/home/rumi/Desktop/projects/my_jobs_env/django-job-portal/jobsapp/api/views/employee.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
/home/rumi/Desktop/projects/my_jobs_env/django-job-portal/jobsapp/api/urls.py changed, reloading.
/home/rumi/Desktop/projects/my_jobs_env/django-job-portal/jobsapp/api/urls.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Bad Request: /api/apply-job/1
Bad Request: /api/apply-job/1
"POST /api/apply-job/1 HTTP/1.1" 400 150
"POST /api/apply-job/1 HTTP/1.1" 400 150
"POST /api/login/ HTTP/1.1" 200 873
"POST /api/login/ HTTP/1.1" 200 873
Bad Request: /api/apply-job/1
Bad Request: /api/apply-job/1
"POST /api/apply-job/1 HTTP/1.1" 400 150
"POST /api/apply-job/1 HTTP/1.1" 400 150
Not Found: /api/applied-jobs/
Not Found: /api/applied-jobs/
"GET /api/applied-jobs/ HTTP/1.1" 404 7208
"GET /api/applied-jobs/ HTTP/1.1" 404 7208
Unauthorized: /api/applied-jobs
Unauthorized: /api/applied-jobs
"GET /api/applied-jobs HTTP/1.1" 401 141
"GET /api/applied-jobs HTTP/1.1" 401 141
"GET /api/applied-jobs HTTP/1.1" 200 3667
"GET /api/applied-jobs HTTP/1.1" 200 3667
